### PR TITLE
Super duper tracing upgrade

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
* Log forwarding mega-upgrade, including context field variables
* Per layer filtering
* Console and export no longer exclusive

## Why?
Much more configurable

## Checklist

1. Closes 

2. How was this tested:
Existing tests and added one

3. Any docs updates needed?

### CODE REVIEW

**Review by cretz — APPROVED** [review]

File: core/src/telemetry/log_export.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        let top_span = span!(Level::INFO, "yayspan", huh = "wat");
+        let _guard = top_span.enter();
+        info!("Whata?");
+        instrumented("hi");
+        info!("Donezo");
+
+        let logs = fetch_global_buffered_logs();
+        for log in logs {
+            println!("{:?}", log);
+        }
```
Comment (bergundy) [inline core/src/telemetry/log_export.rs:181]: Should this be tested?

Comment (bergundy) [reply core/src/telemetry/log_export.rs:181]: That is with assertions instead of printing?

File: core/src/telemetry/mod.rs [hunk 2]
```diff
@@ -255,18 +232,43 @@ pub fn telemetry_init(opts: &TelemetryOptions) -> Result<&'static GlobalTelemDat
...
+                                    .with_filter(EnvFilter::new(filter)),
+                            )
                         }
                     }
-                    Logger::Forward(filter) => {
-                        log::set_max_level(*filter);
-                        globaldat.core_export_logger = Some(CoreExportLogger::new(*filter));
+                    Logger::Forward { filter } => {
+                        let (export_layer, logs_out) = CoreLogExportLayer::new();
+                        globaldat.logs_out = Some(Mutex::new(logs_out));
```
Comment (bergundy) [inline core/src/telemetry/mod.rs:270]: Not new in this PR but is there a way to avoid the mutex here?

**Review by bergundy — APPROVED** [review]

File: core/src/telemetry/mod.rs (hunk 2 above)
Comment (Sushisource) [reply core/src/telemetry/mod.rs:270]: Nope. Tracing implementations need to access self immutably. The locked operation is very, very cheap.

File: core/src/telemetry/log_export.rs (hunk 1 above)
Comment (Sushisource) [reply core/src/telemetry/log_export.rs:181]: Yeah, may as well verify a few things at least. Will update.

**Pull request merged**